File: 7kyu/Python/doubleEveryOther.py
'''
Write a function that doubles every second integer in a list, starting from the left.

Example:

For input array/list :

[1,2,3,4]
the function should return :

[1,4,3,8]
'''

# method 1
def double_every_other(arr):
    result = []
    # initialize iterator
    i = 0
    # for each num in array
    for num in arr:
        # if that number's index is odd
        if i % 2 != 0:
            # add double that num to the result
            result.append(num * 2)
        # else
        else:
            # add the num to result
            result.append(num)
        # incerase iterator
        i += 1
    # return result
    return result

File: 7kyu/JavaScript/doubleEveryOther.js
/*
Write a function that doubles every second integer in a list, starting from the left.

Example:

For input array/list :

[1,2,3,4]
the function should return :

[1,4,3,8]
*/

// Prep
// p: array of numbers
// r: array where each second number is doubled

// method 1: arrow function, ternary operator
const doubleEveryOther = (arr) => arr.map((num, i) => i % 2 !== 0 ? num * 2 : num);

// method 2: long way but more readable
function doubleEveryOther(arr) {
    // map array
    return arr.map((num, i) => {
        // if index is odd (second number is 1st index), double num
        if (i % 2 !== 0) {
            return num * 2
        // else leave as is
        } else {
            return num
        }
    });
};

// e
doubleEveryOther([2, 5, 9, 16]) // [2, 10, 9, 32]